tests/cpp/test_topk_device_func.cpp

```diff
@@ -0,0 +1,324 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <gtest/gtest.h>
+
+#include <tests/cpp/topk_test_helper.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+
+#include <ATen/cuda/CUDAContext.h>
+#include <algorithm>
+#include <cstdint>
+#include <random>
+#include <vector>
+
+namespace nvfuser {
+
+using TopkDeviceFuncTest = NVFuserTest;
+
+template <typename DataT>
+std::vector<DataT> getVector(at::Tensor tensor) {
+  NVF_ERROR_EQ(tensor.dim(), 1);
+  if (tensor.dtype() == at::kBFloat16) {
+    tensor = tensor.to(at::kFloat);
+  }
+  auto cpu_tensor = tensor.cpu();
+  auto total_elements = tensor.size(0);
+  return std::vector<DataT>(
+      cpu_tensor.data_ptr<DataT>(),
+      cpu_tensor.data_ptr<DataT>() + total_elements);
+}
+
+// Helper function to validate topk correctness
+template <typename DataT>
+bool validateTopkOrder(
+    const at::Tensor& input_tensor,
+    const at::Tensor& values_tensor,
+    const at::Tensor& indices_tensor,
+    int k,
```

Comment:
```suggestion
    size_t k,
```
**[Suggested by AI]** The code change replaces the type of the variable `k` from `int` to `size_t`.

**Summary:** This change likely fixes a build error related to using an `int` where a `size_t` was expected. `size_t` is an unsigned integer type guaranteed to be large enough to represent the size of any object in memory, and is commonly used for array indexing and sizes.  The original `int` type might have been too small on some architectures, or the code was explicitly expecting an unsigned type for size/index calculations. This is a common fix when dealing with array lengths or sizes.